src/transaction/log_applier.c

```diff
@@ -7043,142 +7043,177 @@ la_print_log_arv_header (const char *database_name, LOG_ARV_HEADER * hdr, bool v
 }
 
 /*
- * la_log_page_check() - test the transaction log
- *   return: void
- *   log_path: log path
- *   page_num: test page number
+ * la_get_applied_log_info() - 
+ *   return: NO_ERROR or error code
+ *   database_name(in): db name
+ *   log_path(in): real path of log file
+ *   check_replica_info(in): whether replica node execute applyinfo utility
+ *   verbose(in): applyinfo -v option
+ *   applied_final_lsa(in/out): information for delayed applying copied log
+ *
  */
 int
-la_log_page_check (const char *database_name, const char *log_path, INT64 page_num, bool check_applied_info,
-		   bool check_copied_info, bool check_replica_info, bool verbose, LOG_LSA * copied_eof_lsa,
-		   LOG_LSA * copied_append_lsa, LOG_LSA * applied_final_lsa)
+la_get_applied_log_info (const char *database_name, const char *log_path, bool check_replica_info,
+			 bool verbose, LOG_LSA * applied_final_lsa)
 {
   int error = NO_ERROR;
   int res;
-  char *atchar;
-  char active_log_path[PATH_MAX];
   char *replica_time_bound_str;
+  char log_path_buf[PATH_MAX];
 
   assert (database_name != NULL);
   assert (log_path != NULL);
 
-  atchar = (char *) strchr (database_name, '@');
-  if (atchar)
+  LA_HA_APPLY_INFO ha_apply_info;
+  char timebuf[1024];
+
+  if (realpath (log_path, log_path_buf) != NULL)
     {
-      *atchar = '\0';
+      log_path = log_path_buf;
     }
 
-  /* init la_Info */
-  la_init (log_path, 0);
+  la_init_ha_apply_info (&ha_apply_info);
 
-  if (check_applied_info)
+  res = la_get_ha_apply_info (log_path, database_name, &ha_apply_info);
+  if ((res <= 0) || (ha_apply_info.creation_time.date == 0 && ha_apply_info.creation_time.time == 0))
     {
-      LA_HA_APPLY_INFO ha_apply_info;
-      char timebuf[1024];
-
-      la_init_ha_apply_info (&ha_apply_info);
-
-      res = la_get_ha_apply_info (log_path, database_name, &ha_apply_info);
-      if ((res <= 0) || (ha_apply_info.creation_time.date == 0 && ha_apply_info.creation_time.time == 0))
+      error = res;
+      if (res < 0)
 	{
-	  error = res;
-
 	  goto check_applied_info_end;
 	}
+      else
+	{
+	  PRINT_AND_LOG_ERR_MSG (msgcat_message
+				 (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_APPLYINFO, APPLYINFO_MSG_QUERY_RESULTS_FAIL));
+	  goto check_applied_info_end;
```

Comment:
`error` can be set greater than 0. Is it okay for the error code to go to the `check_applied_info_end`?

Reply:
I think it works normal. This condition only can be passed by `res <=0`. 

`(ha_apply_info.creation_time.date == 0 && ha_apply_info.creation_time.time == 0))` is set when `res<=0`

Reply:
Oh, I missed it. you are right.